Driver relocation

mrossinek opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary

This PR tackles the driver and transformer relocating proposed in #166.
Deprecation warnings of the old locations are handled using the `__getattr_` method of the module itself, as proposed by PEP 562 9https://www.python.org/dev/peps/pep-0562/).

### Details and comments

Below are some examples of what this looks like:
```python
In [1]: from qiskit_nature.drivers import QMolecule
<ipython-input-1-e61c84b091f6>:1: DeprecationWarning: QMolecule has been moved to qiskit_nature.drivers.second_quantization.QMolecule
  from qiskit_nature.drivers import QMolecule
/home/oss/Files/Qiskit/.direnv/python-3.9.4/lib64/python3.9/site-packages/pyscf/lib/misc.py:46: H5pyDeprecationWarning: Using default_file_mode other than 'r' is deprecated. Pass the mode to h5py.File() instead.
  h5py.get_config().default_file_mode = 'a'
<ipython-input-1-e61c84b091f6>:1: DeprecationWarning: QMolecule has been moved to qiskit_nature.drivers.second_quantization.QMolecule
  from qiskit_nature.drivers import QMolecule

In [2]: from qiskit_nature.drivers import PySCFDriver
<ipython-input-2-e7cfeeb446a5>:1: DeprecationWarning: PySCFDriver has been moved to qiskit_nature.drivers.second_quantization.PySCFDriver
  from qiskit_nature.drivers import PySCFDriver

In [3]: from qiskit_nature.drivers.second_quantization import HDF5Driver

In [4]:
```

Some notes:
- `DeprecationWarning`s are emitted whenever a class is imported from the old location, but not from the new one.
- `I don't know why the very first time it get's triggered twice.. @manoelmarques do you have an idea what might cause this?
- `Molecule` and `UnitsType` are not second-quantization specific and, thus, remain in their original location

### Open Tasks

- [x] ensure and check that the CI works
- [x] double check that relocating took place in the entire code base
- [x] especially check tutorials
- [x] check that the docs are being rendered properly
- [x] add a reno

mrossinek commented:
As per the documentation (which was the last open task until just now), I built it locally using `make html` which passed. However, it produced 77 warnings which are not all related to this PR. I do not see the same warnings in the CI so I have included them in the details below. Due to Sphinx' output formatting they are little bit hard to read. @woodsp-ibm would you mind taking a look whether these need to be addressed?

<details>

```
% make html
make -C docs html SPHINXOPTS=
make[1]: Entering directory '/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs'
Running Sphinx v4.0.0
making output directory... done
[autosummary] generating autosummary for: apidocs/qiskit_nature.algorithms.pes_samplers.rst, apidocs/qiskit_nature.algorithms.rst, apidocs/qiskit_nature.circuit.library.rst, apidocs/qiskit_nature.circuit.rst, apidocs/qiskit_nature.converters.rst, apidocs/qiskit_nature.converters.second_quantization.rst, apidocs/qiskit_nature.drivers.rst, apidocs/qiskit_nature.drivers.second_quantization.bosonic_bases.rst, apidocs/qiskit_nature.drivers.second_quantization.gaussiand.rst, apidocs/qiskit_nature.drivers.second_quantization.psi4d.rst, ..., apidocs/qiskit_nature.transformers.second_quantization.rst, index.rst, release_notes.rst, tutorials/01_electronic_structure.ipynb, tutorials/02_vibrational_structure.ipynb, tutorials/03_ground_state_solvers.ipynb, tutorials/04_excited_states_solvers.ipynb, tutorials/05_Sampling_potential_energy_surfaces.ipynb, tutorials/06_calculating_thermodynamic_observables.ipynb, tutorials/index.rst
/home/oss/Files/Qiskit/.direnv/python-3.9.4/lib64/python3.9/site-packages/pyscf/lib/misc.py:46: H5pyDeprecationWarning: Using default_file_mode other than 'r' is deprecated. Pass the mode to h5py.File() instead.
  h5py.get_config().default_file_mode = 'a'
[autosummary] generating autosummary for: /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.QiskitNatureError.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.BOPESSampler.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.EigensolverFactory.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.ExcitedStatesEigensolver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.ExcitedStatesSolver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.GroundStateEigensolver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.GroundStateSolver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.MinimumEigensolverFactory.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.NumPyEigensolverFactory.rst, ..., /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.problems.second_quantization.ElectronicStructureProblem.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.problems.second_quantization.VibrationalStructureProblem.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.BOPESSamplerResult.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.EigenstateResult.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.ElectronicStructureResult.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.ActiveSpaceTransformer.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.BaseTransformer.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.FreezeCoreTransformer.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.rst
[autosummary] generating autosummary for: /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.evaluate_operators.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.qubit_converter.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.returns_groundstate.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.solve.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.AdaptVQE.solver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.BOPESSampler.sample.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.EigensolverFactory.get_solver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.ExcitedStatesEigensolver.solve.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.ExcitedStatesEigensolver.solver.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.algorithms.ExcitedStatesSolver.solve.rst, ..., /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.eigenenergies.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.eigenstates.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.formatted.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.groundenergy.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.groundstate.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.num_occupied_modals_per_mode.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.results.VibrationalStructureResult.raw_result.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.ActiveSpaceTransformer.transform.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.BaseTransformer.transform.rst, /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/stubs/qiskit_nature.transformers.second_quantization.FreezeCoreTransformer.transform.rst
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 32 source files that are out of date
updating environment: [new config] 1813 added, 0 changed, 0 removed
scanning /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/releasenotes/notes for current branch release notes
got versions ['0.1.0-32']
executing qiskit_nature.circuit.library.CHC.assign_parameters.CHC.assign_parameters
executing qiskit_nature.circuit.library.CHC.drawrcuit.library.CHC.draw
executing qiskit_nature.circuit.library.CHC.initializelibrary.CHC.initialize
executing qiskit_nature.circuit.library.CHC.tensoruit.library.CHC.tensor
executing qiskit_nature.circuit.library.HartreeFock.assign_parametersFock.assign_parameters
executing qiskit_nature.circuit.library.HartreeFock.drawbrary.HartreeFock.draw
executing qiskit_nature.circuit.library.HartreeFock.initializeHartreeFock.initialize
executing qiskit_nature.circuit.library.HartreeFock.tensorary.HartreeFock.tensor
executing qiskit_nature.circuit.library.PUCCD.assign_parametersUCCD.assign_parameters
executing qiskit_nature.circuit.library.PUCCD.drawuit.library.PUCCD.draw
executing qiskit_nature.circuit.library.PUCCD.initializebrary.PUCCD.initialize
executing qiskit_nature.circuit.library.PUCCD.tensort.library.PUCCD.tensor
executing qiskit_nature.circuit.library.SUCCD.assign_parametersUCCD.assign_parameters
executing qiskit_nature.circuit.library.SUCCD.drawuit.library.SUCCD.draw
executing qiskit_nature.circuit.library.SUCCD.initializebrary.SUCCD.initialize
executing qiskit_nature.circuit.library.SUCCD.tensort.library.SUCCD.tensor
executing qiskit_nature.circuit.library.UCC.assign_parameters.UCC.assign_parameters
executing qiskit_nature.circuit.library.UCC.drawrcuit.library.UCC.draw
executing qiskit_nature.circuit.library.UCC.initializelibrary.UCC.initialize
executing qiskit_nature.circuit.library.UCC.tensoruit.library.UCC.tensor
executing qiskit_nature.circuit.library.UCCSD.assign_parametersCCSD.assign_parameters
executing qiskit_nature.circuit.library.UCCSD.drawuit.library.UCCSD.draw
executing qiskit_nature.circuit.library.UCCSD.initializebrary.UCCSD.initialize
executing qiskit_nature.circuit.library.UCCSD.tensort.library.UCCSD.tensor
executing qiskit_nature.circuit.library.UVCC.assign_parametersUVCC.assign_parameters
executing qiskit_nature.circuit.library.UVCC.drawcuit.library.UVCC.draw
executing qiskit_nature.circuit.library.UVCC.initializeibrary.UVCC.initialize
executing qiskit_nature.circuit.library.UVCC.tensorit.library.UVCC.tensor
executing qiskit_nature.circuit.library.UVCCSD.assign_parametersCCSD.assign_parameters
executing qiskit_nature.circuit.library.UVCCSD.drawit.library.UVCCSD.draw
executing qiskit_nature.circuit.library.UVCCSD.initializerary.UVCCSD.initialize
executing qiskit_nature.circuit.library.UVCCSD.tensor.library.UVCCSD.tensor
executing qiskit_nature.circuit.library.VSCF.assign_parametersVSCF.assign_parameters
executing qiskit_nature.circuit.library.VSCF.drawcuit.library.VSCF.draw
^[[5;2reading sources... [ 76%] stubs/qiskit_nature.circuit.library.VSCF.ecr                                                                                                                  rexecuting qiskit_nature.circuit.library.VSCF.initializeibrary.VSCF.initialize
executing qiskit_nature.circuit.library.VSCF.tensorit.library.VSCF.tensor
executing qiskit_nature.operators.second_quantization.FermionicOpantization.FermionicOp
executing qiskit_nature.operators.second_quantization.SpinOpnd_quantization.SpinOp
reading sources... [100%] tutorials/index
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/bopes_sampler.py:docstring of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler:1: WARNING: duplicate object description of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler, other instance in stubs/qiskit_nature.algorithms.BOPESSampler, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/__init__.py:docstring of qiskit_nature.transformers.second_quantization:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization, other instance in apidocs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/active_space_transformer.py:docstring of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.ActiveSpaceTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/active_space_transformer.py:docstring of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/active_space_transformer.py:docstring of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer.transform:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.ActiveSpaceTransformer.transform, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/base_transformer.py:docstring of qiskit_nature.transformers.second_quantization.base_transformer.BaseTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.BaseTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/base_transformer.py:docstring of qiskit_nature.transformers.second_quantization.base_transformer.BaseTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.base_transformer.BaseTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/base_transformer.py:docstring of qiskit_nature.transformers.second_quantization.base_transformer.BaseTransformer.transform:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.BaseTransformer.transform, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:docstring of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.FreezeCoreTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:docstring of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:docstring of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer.transform:1: WARNING: duplicate object description of qiskit_nature.transformers.second_quantization.FreezeCoreTransformer.transform, other instance in stubs/qiskit_nature.transformers.second_quantization, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs/apidocs/qiskit_nature.transformers.second_quantization.rst: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] tutorials/index
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py:docstring of qiskit_nature.algorithms.ground_state_solvers.adapt_vqe.AdaptVQE.solve:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/bopes_sampler.py:docstring of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/bopes_sampler.py:docstring of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler.sample:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/excited_states_solvers/excited_states_eigensolver.py:docstring of qiskit_nature.algorithms.excited_states_solvers.excited_states_eigensolver.ExcitedStatesEigensolver.solve:11: WARNING: more than one target found for cross-reference 'EigenstateResult': qiskit_nature.results.EigenstateResult, qiskit_nature.results.eigenstate_result.EigenstateResult
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/excited_states_solvers/excited_states_solver.py:docstring of qiskit_nature.algorithms.excited_states_solvers.excited_states_solver.ExcitedStatesSolver.solve:9: WARNING: more than one target found for cross-reference 'EigenstateResult': qiskit_nature.results.EigenstateResult, qiskit_nature.results.eigenstate_result.EigenstateResult
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/ground_state_eigensolver.py:docstring of qiskit_nature.algorithms.ground_state_solvers.ground_state_eigensolver.GroundStateEigensolver.solve:11: WARNING: more than one target found for cross-reference 'EigenstateResult': qiskit_nature.results.EigenstateResult, qiskit_nature.results.eigenstate_result.EigenstateResult
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/ground_state_solver.py:docstring of qiskit_nature.algorithms.ground_state_solvers.ground_state_solver.GroundStateSolver.solve:9: WARNING: more than one target found for cross-reference 'EigenstateResult': qiskit_nature.results.EigenstateResult, qiskit_nature.results.eigenstate_result.EigenstateResult
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/excited_states_solvers/qeom.py:docstring of qiskit_nature.algorithms.excited_states_solvers.qeom.QEOM.solve:12: WARNING: more than one target found for cross-reference 'EigenstateResult': qiskit_nature.results.EigenstateResult, qiskit_nature.results.eigenstate_result.EigenstateResult
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_ucc_factory.VQEUCCFactory:28: WARNING: more than one target found for cross-reference 'UCC': qiskit_nature.circuit.library.UCC, qiskit_nature.circuit.library.ansatzes.ucc.UCC
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_ucc_factory.VQEUCCFactory:28: WARNING: more than one target found for cross-reference 'UCCSD': qiskit_nature.circuit.library.UCCSD, qiskit_nature.circuit.library.ansatzes.uccsd.UCCSD
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_ucc_factory.VQEUCCFactory:31: WARNING: more than one target found for cross-reference 'HartreeFock': qiskit_nature.circuit.library.HartreeFock, qiskit_nature.circuit.library.initial_states.hartree_fock.HartreeFock
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvcc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_uvcc_factory.VQEUVCCFactory:28: WARNING: more than one target found for cross-reference 'UVCC': qiskit_nature.circuit.library.UVCC, qiskit_nature.circuit.library.ansatzes.uvcc.UVCC
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvcc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_uvcc_factory.VQEUVCCFactory:28: WARNING: more than one target found for cross-reference 'UVCCSD': qiskit_nature.circuit.library.UVCCSD, qiskit_nature.circuit.library.ansatzes.uvccsd.UVCCSD
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_uvcc_factory.py:docstring of qiskit_nature.algorithms.ground_state_solvers.minimum_eigensolver_factories.vqe_uvcc_factory.VQEUVCCFactory:31: WARNING: more than one target found for cross-reference 'VSCF': qiskit_nature.circuit.library.VSCF, qiskit_nature.circuit.library.initial_states.vscf.VSCF
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/bopes_sampler.py:docstring of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/bopes_sampler.py:docstring of qiskit_nature.algorithms.pes_samplers.bopes_sampler.BOPESSampler.sample:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.PCAExtrapolator:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/algorithms/pes_samplers/extrapolator.py:docstring of qiskit_nature.algorithms.pes_samplers.extrapolator.Extrapolator.factory:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/puccd.py:docstring of qiskit_nature.circuit.library.ansatzes.puccd.PUCCD:17: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/puccd.py:docstring of qiskit_nature.circuit.library.ansatzes.puccd.PUCCD:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/ucc.py:docstring of qiskit_nature.circuit.library.ansatzes.ucc.UCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/puccd.py:docstring of qiskit_nature.circuit.library.ansatzes.puccd.PUCCD.generate_excitations:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/succd.py:docstring of qiskit_nature.circuit.library.ansatzes.succd.SUCCD:17: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/succd.py:docstring of qiskit_nature.circuit.library.ansatzes.succd.SUCCD:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/ucc.py:docstring of qiskit_nature.circuit.library.ansatzes.ucc.UCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/succd.py:docstring of qiskit_nature.circuit.library.ansatzes.succd.SUCCD.generate_excitations:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/ucc.py:docstring of qiskit_nature.circuit.library.ansatzes.ucc.UCC:66: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/ucc.py:docstring of qiskit_nature.circuit.library.ansatzes.ucc.UCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/uccsd.py:docstring of qiskit_nature.circuit.library.ansatzes.uccsd.UCCSD:6: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/ucc.py:docstring of qiskit_nature.circuit.library.ansatzes.ucc.UCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/uvcc.py:docstring of qiskit_nature.circuit.library.ansatzes.uvcc.UVCC:6: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/uvcc.py:docstring of qiskit_nature.circuit.library.ansatzes.uvcc.UVCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/uvccsd.py:docstring of qiskit_nature.circuit.library.ansatzes.uvccsd.UVCCSD:6: WARNING: more than one target found for cross-reference 'SecondQuantizedOp': qiskit_nature.operators.second_quantization.SecondQuantizedOp, qiskit_nature.operators.second_quantization.second_quantized_op.SecondQuantizedOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/circuit/library/ansatzes/uvcc.py:docstring of qiskit_nature.circuit.library.ansatzes.uvcc.UVCC.excitation_ops:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/base_driver.py:docstring of qiskit_nature.drivers.second_quantization.base_driver.BaseDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/bosonic_driver.py:docstring of qiskit_nature.drivers.second_quantization.bosonic_driver.BosonicDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/fcidumpd/fcidumpdriver.py:docstring of qiskit_nature.drivers.second_quantization.fcidumpd.fcidumpdriver.FCIDumpDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/fermionic_driver.py:docstring of qiskit_nature.drivers.second_quantization.fermionic_driver.FermionicDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/gaussiand/gaussiandriver.py:docstring of qiskit_nature.drivers.second_quantization.gaussiand.gaussiandriver.GaussianDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/gaussiand/gaussian_forces_driver.py:docstring of qiskit_nature.drivers.second_quantization.gaussiand.gaussian_forces_driver.GaussianForcesDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/gaussiand/gaussian_log_driver.py:docstring of qiskit_nature.drivers.second_quantization.gaussiand.gaussian_log_driver.GaussianLogDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/gaussiand/gaussian_log_driver.py:docstring of qiskit_nature.drivers.second_quantization.gaussiand.gaussian_log_driver.GaussianLogDriver.run:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/psi4d/psi4driver.py:docstring of qiskit_nature.drivers.second_quantization.psi4d.psi4driver.PSI4Driver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/pyquanted/pyquantedriver.py:docstring of qiskit_nature.drivers.second_quantization.pyquanted.pyquantedriver.PyQuanteDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/drivers/second_quantization/pyscfd/pyscfdriver.py:docstring of qiskit_nature.drivers.second_quantization.pyscfd.pyscfdriver.PySCFDriver:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/direct_mapper.py:docstring of qiskit_nature.mappers.second_quantization.direct_mapper.DirectMapper:3: WARNING: more than one target found for cross-reference 'VibrationalOp': qiskit_nature.operators.second_quantization.VibrationalOp, qiskit_nature.operators.second_quantization.vibrational_op.VibrationalOp
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/mappers/second_quantization/qubit_mapper.py:docstring of qiskit_nature.mappers.second_quantization.qubit_mapper.QubitMapper.mode_based_mapping:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/operators/second_quantization/spin_op.py:docstring of qiskit_nature.operators.second_quantization.spin_op.SpinOp:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/active_space_transformer.py:docstring of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer.transform:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:docstring of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer.transform:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/active_space_transformer.py:docstring of qiskit_nature.transformers.second_quantization.active_space_transformer.ActiveSpaceTransformer.transform:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:docstring of qiskit_nature.transformers.second_quantization.freeze_core_transformer.FreezeCoreTransformer.transform:: WARNING: more than one target found for cross-reference 'QiskitNatureError': qiskit_nature.QiskitNatureError, qiskit_nature.exceptions.QiskitNatureError
generating indices... genindex py-modindex done
copying notebooks ... [100%] tutorials/06_calculating_thermodynamic_observables.ipynb
highlighting module code... [100%] qiskit_nature.transformers.second_quantization.freeze_core_transformer
writing additional pages... search done
copying images... [100%] _build/doctrees/nbsphinx/tutorials_06_calculating_thermodynamic_observables_24_0.png
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 77 warnings.

The HTML pages are in _build/html.
make[1]: Leaving directory '/home/oss/Files/Qiskit/src/qiskit-nature/feature/driver-relocation/docs'
make html  245.51s user 19.27s system 55% cpu 8:01.21 total
```
</details>

mrossinek commented:
@woodsp-ibm @pbark Can we please get some movement on a review here?

woodsp-ibm commented:
I am wondering, given that there is further driver refactoring to be done, such as splitting out the MoleculeDriver, etc whether it would be simpler/easier/cleaner to just keep a copy of the code where it is and add deprecation messages to it, and have the new driver code in the new location simply just be whatever is the new refactored code, initially just a simple copy of what is there now but which can be refactored and not have to have it keep the deprecated paths for Molecule etc in this new code. We can keep the driver tests and make a copy too in the new location - eventually the old code and old tests would just simply then be removed leaving only the new code. If we were just relocating the drivers what is here is undoubtedly the better approach, but with the intended refactor of the relocated code I am not so sure.

mrossinek commented:
Thanks for the input. I did as you suggested.

Unfortunately, in the process, some odd merge commit happened because I was out-of-sync with the public branch. I rebased this branch but now Laurin appears as a co-author of one of the merged (and now rebased) commits. I am unable to get rid of this artifact.

mrossinek commented:
Since the number of changes in this PR has grown *a lot*, here is a quick summary of what this PR does:

- move all `qiskit_nature/transformers` to `qiskit_nature/transformers/second_quantization`
- deprecate all classes in `qiskit_nature/drivers`
- duplicate all classes from `qiskit_nature/drivers` into `qiskit_nature/drivers/second_quantization` (except `Molecule` and `UnitsType`)
- updates all imports to the new locations
- updates all docstrings to the new locations
- move all `test/transformers` to `test/transformers/second_quantization`
- duplicates all `test/drivers` into `test/drivers/second_quantization` in-line with the above
- updates all `get_resource_path` calls in the test suite to the new locations

The reason for the different approach with the drivers was suggested by @woodsp-ibm [here](https://github.com/Qiskit/qiskit-nature/pull/173#issuecomment-847981378).

woodsp-ibm commented on .gitignore:
```diff
@@ -20,8 +20,8 @@ __pycache__/
 #Allow
-!qiskit_nature/drivers/gaussiand/gauopen/*.so
-!qiskit_nature/drivers/gaussiand/gauopen/*.pyd
+!qiskit_nature/drivers/second_quantization/gaussiand/gauopen/*.so
+!qiskit_nature/drivers/second_quantization/gaussiand/gauopen/*.pyd
```
Since the drivers have been kept in their old place for now too, we should have both of the sections in there right, (ie do not delete the original lines 23/24)

mrossinek replied:
Thanks for spotting this! Reverting the moving was a rather manual process in some cases because I did not want to revert the entire commit. I missed this aspect. Will fix asap

woodsp-ibm commented on MANIFEST.in:
```diff
@@ -1,7 +1,7 @@
 include requirements.txt
 recursive-include qiskit_nature *.txt
-graft qiskit_nature/drivers/gaussiand/gauopen
+graft qiskit_nature/drivers/second_quantization/gaussiand/gauopen
```
Similar comment here to above - we need both I think

woodsp-ibm commented on docs/tutorials/07_leveraging_qiskit_runtime.ipynb:
```diff
@@ -39,7 +39,7 @@
    "outputs": [],
    "source": [
-    "from qiskit_nature.drivers import PySCFDriver, UnitsType, Molecule\n",
+    "from qiskit_nature.drivers.second_quantization import PySCFDriver, UnitsType, Molecule\n",
```
Shouldn't this line look more like was done with the other notebooks i.e.
```
from qiskit_nature.drivers import UnitsType, Molecule
from qiskit_nature.drivers.second_quantization import PySCFDriver
```
where UnitsType and Molecule come from root of drivers

mrossinek replied:
Yes, thanks!

woodsp-ibm commented on qiskit_nature/drivers/gaussiand/gaussiandriver.py:
```diff
@@ -69,6 +69,14 @@ def __init__(
         Raises:
             QiskitNatureError: Invalid Input
         """
+        warnings.warn(
```
If we split out the Molecule aspect from the Native aspect (current intent) perhaps we might want to tailor this later after that refactoring (and likewise deprecation messages in the other drivers) to indicate the new MoleculeDriver if they are constructing the driver using that, otherwise the native driver as it does presently.

mrossinek replied:
I agree. I will link this to #165 such that we don't forget.

woodsp-ibm commented:
While we maintain the deprecated code and the tests that ensure it still working we should add to those tests, maybe in the setup and teardown, disabling and renabling the deprecation messages since we know they will be generated and we want to ignore these. Reason being that the deprecation messages are collected by CI and the goal before release is not to be using deprecated code hence, aside from 3rd party dependents that we can't fix, we should not have any other deprecation messages there - trying to wade through a bunch that we know will be there, we can easily miss ones we should be fixing, hence when testing code we know to be deprecated we normally just suppress the messages - and turn them back on after. It can be done more localized in the unit test but that may not be so practical in this case.

woodsp-ibm commented on test/nature_test_case.py:
```diff
@@ -42,7 +42,7 @@ class QiskitNatureTestCase(unittest.TestCase, ABC):
         warnings.filterwarnings("default", category=DeprecationWarning)
         warnings.filterwarnings("ignore", category=DeprecationWarning, module="pyscf")
-        warnings.filterwarnings(action='ignore', category=DeprecationWarning, module='.*drivers.*')
+        warnings.filterwarnings(action="ignore", category=DeprecationWarning, module=".*drivers.*")
```
Doesn't this end up suppressing all such warnings from any code under drivers, including the new relocated ones. If we ended up using any deprecated code there we would want to fix it and hence see any messages in the build output. Should it have another line to turn things on again if its drivers.second_quantization.*? 

I note that there are still deprecation messages appearing in the CI log from HFMethod and InitialGuess enums

woodsp-ibm reviewed (approved):
Thanks Manoel for polishing up the deprecated logic - a bit more involved then to handle the enums. Looks good to me.

mrossinek commented:
Thanks a lot, @manoelmarques! :slightly_smiling_face:
